uniconnect-epfl/uniconnect
# feat: add contact by qr

# What I did
- Finished the qr code scanning logic and implemented a rudimentary "add a contact" screen. 
- We cannot add a contact yet because the database does not keep track of contacts. 
- When scanning from inside the app, it is checked that the user exists and is not the same user as the one scanning, however when scanning from outside the app these cases aren't detected. I'm thinking of implementing these checks in another PR.

# How I did it
- Modified the QrScanScreen to parse the parsed link and act depending on the link.
- Added the dependency `use-debounce` and used it to make a debouncer for qr code scanning.
- Implemented the AddContactScreen.
- Had to modify a few tests to adapt for the changes.

# How to verify it
Try to scan this qr code with the scanner of the app and then with the scanner of your phone. You can also try to scan your own qr code, the app should tell you it's yourself.

<img src="https://github.com/uniconnect-epfl/uniconnect/assets/93329823/023d5ce3-f5ae-48f4-922e-cf51c617d156"  height="500">

# Demo video

https://github.com/uniconnect-epfl/uniconnect/assets/93329823/5f63ee16-53e0-4a1c-ae9f-cf9501f195f5

# Pre-merge checklist
The changes I have introduced:
